### PR TITLE
Refactor functional tests to use AEPTestUtils

### PR DESCRIPTION
**Pull request by timkimadobe**
> [!NOTE]
> The Podfile dependency currently points to a branch of the testutils repo - this can be updated to point to a particular `tag` (once release tag is created) using the syntax, ex: `pod 'AEPTestUtils', :git => 'https://github.com/adobe/aepsdk-testutils-ios.git', :tag => '1.0.0-beta'`

## Description

This PR brings in the `AEPTestUtils` library and replaces existing local versions of test utilities with the following changes:
1. Import AEPTestUtils in all dependent classes
2. Refactor individual property assertions with JSON comparison

## Questions for reviewers
I had some questions about the existing test case logic for the following cases in `ConsentFunctionalTests.swift`:
`testBootup_CachedConsentsExist_ConfigDefaultExist_DefaultsUpdated`
Intentional that second shared state doesnt validate `share`?

`testUpdateConsentHappyIgnoresMetadataDate`
Is the strict equality of timestamp enough? Does `metadataDate` also need to be negatively checked? Or compared against `event.timestamp` to not be equal?

`testValidResponseWithEmptyExistingConsentsIgnoresExtraneous`
Shared state has extra key:
```json
"personalize": {
  "content": {
    "val": "y"
  }
}
```
Which is not validated, is this desired?

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPEdgeConsent.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -309,23 +300,12 @@
 		21A9E2B625D1EDB90089690E /* Helpers */ = {
 			isa = PBXGroup;
 			children = (
-				21E80A3025BF6EAD00011C18 /* TestableExtensionRuntime.swift */,
-				21A9E2AA25D1ED910089690E /* EventHub+Testable.swift */,
 				21A9E2B025D1EDB10089690E /* MockExtension.swift */,
 				2158820825E57C4C005B830B /* Dictionary+Flatten.swift */,
```
Comment (emdobrin) [inline AEPEdgeConsent.xcodeproj/project.pbxproj:304]: where are these two helpers still used or can we remove 🙏 ?

File: Tests/FunctionalTests/ConsentPublicAPITests.swift [hunk 2]
```diff
@@ -67,16 +69,28 @@ class ConsentPublicAPITests: XCTestCase {
         let expectation = XCTestExpectation(description: "updateConsents should dispatch an event with correct payload")
         expectation.assertForOverFulfill = true
         EventHub.shared.getExtensionContainer(MockExtension.self)?.registerListener(type: EventType.edgeConsent, source: EventSource.updateConsent) { event in
+            // Verify - Consents in update event should be equal
             let dispatchedConsents = ConsentPreferences.from(eventData: event.data!)
-            let equal = NSDictionary(dictionary: consents.asDictionary()!).isEqual(to: (dispatchedConsents?.asDictionary())!) // consents in update event should be equal
-            XCTAssertTrue(equal)
+
+            let expectedConsentsJSON = #"""
```
Comment (emdobrin) [inline Tests/FunctionalTests/ConsentPublicAPITests.swift:75]: consents is now unused, let's remove

**Comment by emdobrin** [comment]
In the attempt to answer your questions:
1. > testBootup_CachedConsentsExist_ConfigDefaultExist_DefaultsUpdated Intentional that second shared state doesnt validate share?

    I don't think it was intentional, you can include that check since we are expecting the responseContent and the sharedState to have same data.

2. > testUpdateConsentHappyIgnoresMetadataDate
Is the strict equality of timestamp enough? Does metadataDate also need to be negatively checked? Or compared against event.timestamp to not be equal?

    The test is checking the metadata value is set internally as the event timestamp (the time when the update has been recorded with the sdk). some negative testing may include passing in metadata as nil, non-string values and sdk handles gracefully, or we can test that consecutive updates take into effect latest event timestamp 

3. > testValidResponseWithEmptyExistingConsentsIgnoresExtraneous

    We can assertEqual on this, I believe the test meant to check that extraneous keys are ignored, but they are not excluded. there is no validation on known keys or such.

File: Tests/FunctionalTests/ConsentPublicAPITests.swift (hunk 2 above)
Comment (emdobrin) [reply Tests/FunctionalTests/ConsentPublicAPITests.swift]: the `consents` constant above does not seem to be used

File: AEPEdgeConsent.xcodeproj/project.pbxproj (hunk 1 above)
Comment (emdobrin) [reply AEPEdgeConsent.xcodeproj/project.pbxproj]: Is it generic enough and are we using it in multiple places?

File: Tests/FunctionalTests/ConsentFunctionalTests.swift [hunk 3]
```diff
@@ -35,238 +36,353 @@ class ConsentFunctionalTests: XCTestCase {
...
+            }
+          }
+        }
+        """
 
-        let flatEdgeUpdateEvent = mockRuntime.dispatchedEvents.last?.data?.flattening()
-        XCTAssertEqual("n", flatEdgeUpdateEvent?["consents.collect.val"] as? String)
-        XCTAssertNil(flatEdgeUpdateEvent?["consents.adID.val"]) // edge event should only contain net new consents from buildSecondUpdateConsentEvent()
-        XCTAssertEqual(secondUpdateConsentEvent.timestamp.iso8601UTCWithMillisecondsString, flatEdgeUpdateEvent?["consents.metadata.time"] as? String)
+        assertExactMatch(expected: expectedEdgeEvent, actual: edgeUpdateEvent, pathOptions: KeyMustBeAbsent(paths: "consents.adID.val"))
```
Comment (emdobrin) [inline Tests/FunctionalTests/ConsentFunctionalTests.swift:319]: same comment as above, use assertEqual wherever you can

File: Tests/FunctionalTests/ConsentPublicAPITests.swift (hunk 2 above)
Comment (emdobrin) [reply Tests/FunctionalTests/ConsentPublicAPITests.swift]: Ok, you can move it right before the usage

**Review by kevinlind — COMMENTED** [review]
These changes look good to me, as long as you make the changes Emilia suggested. I agree with her that it appears most of the tests could use `assertEqual`.

**Review by emdobrin — APPROVED** [review]

**Review by kevinlind — APPROVED** [review]

**Pull request merged**